tests/Microsoft.Azure.Batch.SoftwareEntitlement.Tests/FakeTokenPropertyProvider.cs

```diff
@@ -8,57 +8,72 @@ namespace Microsoft.Azure.Batch.SoftwareEntitlement.Tests
 {
     public sealed class FakeTokenPropertyProvider : ITokenPropertyProvider
     {
-        public static FakeTokenPropertyProvider CreateValid()
+        public static FakeTokenPropertyProvider CreateValid(DateTimeOffset? issuedAt = null)
```

Comment:
I was assuming this was data that had to change between test runs. Because the method is used to create a "valid" token property provider I figured the current time would need to fall between not-before and not-after, and was trying to encourage consumers (the tests themselves) to explicitly specify time-related data affecting whether tests pass, rather than having it buried in this class. But by making it optional it didn't have the intended effect.

I think the source of my confusion is that I chose an ambiguous method name "CreateValid". Does it mean the token provider has no errors, or does it mean the resulting token can be verified successfully?

I'll do a bit of renaming and add doc comments to try and make it clearer.